lib/tictactoe.rb

```diff
@@ -39,6 +50,44 @@ def show_board(board_spaces)
   end
 
 
+  def check_for_winner?(board)
+    winner_symbol = ""
+    
+    for win_combo in WIN_COMBOS do
+      row = Array.new
+      for index in win_combo do 
+        row.append(board[win_combo[index]])
+      end
+       if (row.all? { |each| each == row[0]})
+          if row[0] != " "
+            winner_symbol = row[0]
+          end
+       end
+
+      # old code 
+      # win_index_1 = win_combo[0]
+      # win_index_2 = win_combo[1]
+      # win_index_3 = win_combo[2]
+
+      # winning_position_1 = board[win_index_1]
+      # winning_position_2 = board[win_index_2] 
+      # winning_position_3 = board[win_index_3] 
+
+      # if (winning_position_1 == winning_position_2) && (winning_position_2 == winning_position_3)
+      #   if winning_position_1 != " "
+      #     winner_symbol = winning_position_1
+      #   end
+      # end
+    
+    
+    #for loop end
+    end
+    winner_symbol != "" ? winner_symbol : false 
```

Comment:
Ideally, your methods should not return different types. Here you either return a string of false. The Ruby way would be to return `nil` if a function is unable to return a value being asked for. And false is used as a return from a function that check true or false.

Reply:
Understood. Will change

Reply:
changed it to return to nil,  I like this because I can keep the string of the winner symbol return so i can use it to display who won in the app 
Compared to switching the method to return just truth or false and losing the ability to display who won.